VERSIONING.md

```diff
@@ -0,0 +1,55 @@
+Rocket Chip Versioning Strategy ** All Strawman to be discussed**
+===============================
+
+Version Naming
+--------------
+
+* Rocket Chip uses a Calendar Versioning release strategy (https://calver.org/).
+* The format is YYYY.MM.DD.MINOR
+* MINOR changes include:
+  * API Additions
+  * Bug Fixes (which can result in changes in the emitted hardware considered bugs)
+* MINOR changes do NOT include:
+  * Changes in top-level I/Os for < level of hierarchy >
+  * Backwards incompatible changes in the API (see below)
+
+API
+---
+
+The Rocket-Chip API is: 
+  * All public Scala methods and classes
+  * All public Scala methods and classes of included Scala libraries (hardfloat, CDE, etc)
+  * The command line used to invoke the generator
+  * All Verilog interfaces (I/Os and parameterization) for verilog files
+  * C code...?
+  * The version(s) of Chisel, FIRRTL, rocket-tools pointed to by rocket-chip.
+  * Any code that generates IO(s) of any Module
+  * The cmd line interface to scripts like vlsi-mem-gen
+  * `public val`s within `LazyModule`s which translate to diplomatic nodes (?? -- this could go either way)
```

Comment:
How about the Makefiles/Makefrags?

Also, what about the Wake files and wit-manifest.json?

Reply:
> How about the Makefiles/Makefrags?
> Also, what about the Wake files

Good point. Do you have a suggestion? How is this dealt with in other projects?

> and wit-manifest.json?

To me this is like asking "what about the submodules" which I tried to capture as well. Is there something special about wit-manifest.json that should be captured differently than submodules? In particular, shoudl we be explicit about changes to the version of wit/wake needed to deal with this repo (for example, if we make a change to this repo that requires you to bump to a new major version of wit/wake, in my mind that should be a MAJOR change).

Not sure the right classification if we make a change that requires you to bump to a new minor version of wit/wake. I think this is sort of what @grebe was discussing above RE chisel/firrtl, but it's a little inverted because rocket-chip does not *provide* wit/wake wheras rocket-chip is often used as the *provider* of chisel/firrtl to projects which use rocket-chip as a library.

Reply:
> > How about the Makefiles/Makefrags?
> > Also, what about the Wake files
> 
> Good point. Do you have a suggestion? How is this dealt with in other projects?

In my experience, most other software projects do not attempt to rely on the build scripts of their dependencies at all, and they are not part of the "interface". However, I was under the impression that some UCB folks or other open source users may be relying on them.

> Is there something special about wit-manifest.json that should be captured differently than submodules / the repos it points to?

I agree that it's mostly the same as the Git submodules. However, one big technical difference is that the `wit-manifest.json` describes a `>=` relationship, not an `==` relationship, so if in the `wit-manifest.json` you say that you depend on version A of a library, you are really saying that you depend on A or any version that has A as an ancestor. This does make it a little bit easier on versioning since users are allowed to use any version of the dependency that is newer.

Reply:
FWIW, we (chipyard & firesim) do not use Rocket Chip's checked-in makefrags but do use the generated ones.

Reply:
I added Makefiles/Makefrags to the "NOT part of the API" section, and then tried to clarify the wit-manifest dependencies behavior. Can you take a look @richardxia to see if what I wrote in https://github.com/chipsalliance/rocket-chip/pull/2284/files#diff-baa362176b63e6912fc6721235a68502R47
helps answer your question?

Reply:
That answers my question, but it sounds like chipyard and firesim use the generated Makefiles.

FWIW, SiFive does not use any of rocket-chip Makefiles/Makefrags/generated Makefiles internally, either.